leftwm-core/src/handlers/focus_handler.rs

```diff
@@ -151,6 +151,17 @@ impl State {
         }
     }
 
+    /// Finds the managed window for a [`leftwm_core::models::window::WindowHandle`].
+    pub fn find_window(&self, handle: &WindowHandle) -> Option<Window> {
+        // Find the handle in our managed windows.
+        let found: &Window = self.windows.iter().find(|w| &w.handle == handle)?;
+        // Docks don't want to get focus. If they do weird things happen. They don't get events...
+        if !found.is_managed() {
+            return None;
+        }
+        Some(found.clone())
+    }
```

Comment:
I'm sure we can add lifetimes to this so it returns `Option<&Window>`.

Edit: Or just re-integrate it below when removing the new event.

Reply:
Well the original code was also doing a clone, so we will need to rework regardless if we want to use a lifetime instead and avoid the clone.

Reply:
In the mean time I just reverted the change to use the old event as suggested.